tools/codegen/pkg/interfacegen/generator_test.go

```diff
@@ -15,159 +15,89 @@
 package interfacegen
 
 import (
+	"bytes"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"os"
-	"os/exec"
-	"path"
-	"path/filepath"
 	"testing"
 )
 
-func TestWellKnownTemplate(t *testing.T) {
+type logFn func(string, ...interface{})
+
+// TestBazelGeneration uses the outputs files generated via the bazel rule
+// for testdata:generated_files and compares them against the golden files.
+func TestBazelGeneration(t *testing.T) {
 	tests := []struct {
-		src      string
-		baseline string
+		name, got, want string
 	}{
-		{"testdata/MetricTemplate.proto", "testdata/MetricTemplateProcessorInterface.go.baseline"},
-		{"testdata/MetricTemplate.proto", "testdata/MetricTemplateProcessorInterface.go.baseline"},
-		{"testdata/QuotaTemplate.proto", "testdata/QuotaTemplateProcessorInterface.go.baseline"},
-		{"testdata/LogTemplate.proto", "testdata/LogTemplateProcessorInterface.go.baseline"},
-		{"testdata/ListTemplate.proto", "testdata/ListTemplateProcessorInterface.go.baseline"},
-		{"testdata/NestedMessage.proto", "testdata/NestedMessageProcessor.go.baseline"},
-		{"testdata/ErrorTemplate.proto", "testdata/ErrorTemplate.baseline"},
+		{"Metrics", "testdata/metric_template_library_processor.gen.go", "testdata/MetricTemplateProcessorInterface.golden.go"},
+		{"Quota", "testdata/quota_template_library_processor.gen.go", "testdata/QuotaTemplateProcessorInterface.golden.go"},
+		{"Logs", "testdata/log_template_library_processor.gen.go", "testdata/LogTemplateProcessorInterface.golden.go"},
+		{"Lists", "testdata/list_template_library_processor.gen.go", "testdata/ListTemplateProcessorInterface.golden.go"},
+		{"Nested Message", "testdata/nested_message_library_processor.gen.go", "testdata/NestedMessageProcessorInterface.golden.go"},
```

Comment:
I've updated to use *.descriptor_set files and direct calls to Generate(). I've also updated the `bin/bazel_to_go.py` to put everything in the right place for gogland/IDE/go testing.  Look for it in the next commit I push up.